pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -142,7 +137,7 @@ private void receiveMessageFromConsumer(ConsumerImpl consumer) {
     }
 
     private void resumeReceivingFromPausedConsumersIfNeeded() {
-        if (incomingMessages.size() <= sharedQueueResumeThreshold && !pausedConsumers.isEmpty()) {
+        if (incomingMessages.size() < maxReceiverQueueSize && !pausedConsumers.isEmpty()) {
```

Comment:
ok.. so, here issue is: this condition ```incomingMessages.size() <= sharedQueueResumeThreshold``` will prevent ```pausedConsumers``` to add messages into shared queue. and it will allow [unpaused](https://github.com/yahoo/pulsar/blob/master/pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/PartitionedConsumerImpl.java#L131) consumer to keep adding messages to queue if client is receiving messages at the same rate. 

So, to create a fair chance: if any paused-consumer is present then rather allowing unpaused consumer adding messages, add it to the ```pausedConsumers``` list and let it to be picked up from ```pausedConsumers```  in round-robin.
Updated PR with the change.